yt_dlp/extractor/vidyard.py

```diff
@@ -0,0 +1,156 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, traverse_obj
+
+
+class VidyardBaseInfoExtractor(InfoExtractor):
+
+    def _get_formats_and_subtitle(self, video_source, video_id):
+        video_source = video_source or {}
+        formats, subtitles = [], {}
+        for key, value in video_source.items():
+            if key == 'hls':
+                for video_hls in value:
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(video_hls.get('url'), video_id, headers={
+                        'referer': 'https://play.vidyard.com/'
+                    })
+                    formats.extend(fmts)
+                    self._merge_subtitles(subs, target=subtitles)
+            else:
+                formats.extend({
+                    'url': video_mp4.get('url'),
+                    'ext': 'mp4'
+                } for video_mp4 in value)
+
+        return formats, subtitles
+
+    def _get_direct_subtitle(self, caption_json):
+        subs = {}
+        for caption in caption_json:
+            subs.setdefault(caption.get('language') or 'und', []).append({
```

Comment:
Not specifically relevant here, but perhaps it should be documented that the Undetermined ISO 639 language code should be used if an extractor can't find (or deduce from the site or page) the language for a subtitle.